### PR TITLE
Clarify Proc#call vs yield, add explanation

### PR DESCRIPTION
**Pull request by nateberkopec**

### CODE REVIEW

File: code/proc-and-block/proc-call-vs-yield.rb [hunk 1]
```diff
@@ -8,6 +8,10 @@ def slow2 &block
   yield
 end
 
+def slow3 &block
```
Comment (ixti) [inline code/proc-and-block/proc-call-vs-yield.rb]: Please wrap argument with brackets.

File: README.md [hunk 2]
```diff
@@ -733,23 +733,29 @@ Comparison:
                Block:    47914.3 i/s - 1.14x slower
 ```
 
-##### `Proc#call` vs `yield` [code](code/proc-and-block/proc-call-vs-yield.rb)
+##### `Proc#call` and block arguments vs `yield`[code](code/proc-and-block/proc-call-vs-yield.rb)
+
+[In MRI Ruby, block arguments are converted to Procs, which incurs a heap allocation](https://www.omniref.com/ruby/2.2.0/symbols/Proc/yield?#annotation=4087638&line=711).
```
Comment (ixti) [inline README.md]: IMO it worth to make link text a bit smaller. :D

``` markdown
In MRI Ruby, block arguments [are converted to Procs](...), which incurs...
```

Comment (JuanitoFatas) [inline README.md]: :+1:

File: code/proc-and-block/proc-call-vs-yield.rb (hunk 1 above)
Comment (JuanitoFatas) [inline code/proc-and-block/proc-call-vs-yield.rb]: :+1:

**Comment by JuanitoFatas** [comment]
@nateberkopec Thank you for the explanations!!! :bow:

**Pull request closed without merging**